src/osx/webview_webkit.mm

```diff
@@ -231,93 +219,49 @@ @interface WebViewCustomProtocol : NSURLProtocol
     if ( !m_webView )
         return;
 
-    [[m_webView mainFrame] stopLoading];
-}
-
-bool wxWebViewWebKit::CanGetPageSource() const
-{
-    if ( !m_webView )
-        return false;
-
-    WebDataSource* dataSource = [[m_webView mainFrame] dataSource];
-    return ( [[dataSource representation] canProvideDocumentSource] );
+    [m_webView stopLoading];
 }
 
 wxString wxWebViewWebKit::GetPageSource() const
 {
-
-    if (CanGetPageSource())
-    {
-        WebDataSource* dataSource = [[m_webView mainFrame] dataSource];
-        wxASSERT (dataSource != nil);
-
-        id<WebDocumentRepresentation> representation = [dataSource representation];
-        wxASSERT (representation != nil);
-
-        NSString* source = [representation documentSource];
-        if (source == nil)
-        {
-            return wxEmptyString;
-        }
-
-        return wxCFStringRef::AsString( source );
-    }
-
-    return wxEmptyString;
+    wxString text;
+    const_cast<wxWebViewWebKit*>(this)->RunScript("document.documentElement.outerHTML;", &text);
+    return text;
 }
 
-bool wxWebViewWebKit::CanIncreaseTextSize() const
+void wxWebViewWebKit::Print()
 {
-    if ( !m_webView )
-        return false;
 
-    if ([m_webView canMakeTextLarger])
-        return true;
-    else
-        return false;
-}
+    // TODO: allow specifying the "show prompt" parameter in Print() ?
+    bool showPrompt = true;
 
-void wxWebViewWebKit::IncreaseTextSize()
-{
     if ( !m_webView )
         return;
 
-    if (CanIncreaseTextSize())
-        [m_webView makeTextLarger:(WebView*)m_webView];
-}
-
-bool wxWebViewWebKit::CanDecreaseTextSize() const
-{
-    if ( !m_webView )
-        return false;
+    // As of macOS SDK 10.15 no offical printing API is available for WKWebView
+    // Try if the undocumented printOperationWithPrintInfo: is available and use it
+    // to create a printing operation
+    // https://bugs.webkit.org/show_bug.cgi?id=151276
+    SEL printSelector = @selector(printOperationWithPrintInfo:);
+    if (![m_webView respondsToSelector:printSelector])
+        printSelector = nil;
 
-    if ([m_webView canMakeTextSmaller])
-        return true;
-    else
-        return false;
-}
-
-void wxWebViewWebKit::DecreaseTextSize()
-{
-    if ( !m_webView )
+    if (!printSelector)
+    {
+        wxLogError(_("Printing selector is not available"));
```

Comment:
I'm afraid this might be rather puzzling too, especially as this is meant to the user and not the developer. I'd suggest
```suggestion
        wxLogError(_("Printing is not supported by the system web control"));
```